sound/soc/sof/sof-audio.c

```diff
@@ -465,7 +466,7 @@ int sof_machine_check(struct snd_sof_dev *sdev)
 	if (!mach)
 		return -ENOMEM;
 
-	mach->drv_name = "sof-nocodec";
+	mach->drv_name = "snd_sof_nocodec_drv";
```

Comment:
I don't get why you need the driver name?

Reply:
this is required for the ignore_machine logic in the core.

Reply:
sorry, you lost me here. why do we need the 'ignore-machine' logic for a nocodec card.

Reply:
@plbossart it doesnt matter what card , we use the ignore_machine check here: https://github.com/thesofproject/linux/blob/41fea6ee2ca94d2e69377f3d102f0bb6874d2511/sound/soc/soc-core.c#L1643 
to set the stream name for the DAI links here:
https://github.com/thesofproject/linux/blob/41fea6ee2ca94d2e69377f3d102f0bb6874d2511/sound/soc/soc-core.c#L1702

If this doesnt happen, we'd fail while loading the DAI during topology loading: https://github.com/thesofproject/linux/blob/41fea6ee2ca94d2e69377f3d102f0bb6874d2511/sound/soc/sof/topology.c#L1390

Reply:
but thinking more about it, maybe we can changethe nocodec machine driver to add the stream names. I'll have to test this out

Reply:
fixed now